src/common/runtime/standalone.ts

```diff
@@ -22,18 +22,71 @@ window.onbeforeunload = () => {
 
 let haveSomeResults = false;
 
-const runnow = optionEnabled('runnow');
-const debug = optionEnabled('debug');
-const unrollConstEvalLoops = optionEnabled('unroll_const_eval_loops');
+interface StandaloneOptionInfo {
+  description: string;
+  parser?: (key: string) => boolean | string;
+  selectValueDescriptions?: { value: string; description: string }[];
+}
+
+interface StandaloneOptions {
+  runnow: boolean;
+  worker: boolean;
+  debug: boolean;
+  unrollConstEvalLoops: boolean;
+  powerPreference: string;
+}
+
+type StandaloneOptionsInfos = Record<keyof StandaloneOptions, StandaloneOptionInfo>;
+
+const optionsInfo: StandaloneOptionsInfos = {
+  runnow: { description: 'run immediately on load' },
+  worker: { description: 'run in a worker' },
+  debug: { description: 'show more info' },
+  unrollConstEvalLoops: { description: 'unroll const eval loops in WGSL' },
+  powerPreference: {
+    description: 'set default powerPreference for some tests',
+    parser: optionString,
+    selectValueDescriptions: [
+      { value: '', description: 'default' },
+      { value: 'low-power', description: 'low-power' },
+      { value: 'high-performance', description: 'high-performance' },
+    ],
+  },
+};
+
+function camelCaseToSnakeCase(id: string) {
+  return id
+    .replace(/(.)([A-Z][a-z]+)/g, '$1_$2')
+    .replace(/([a-z0-9])([A-Z])/g, '$1_$2')
+    .toLowerCase();
+}
 
+function getOptionsInfoFromSearchParameters(
+  optionsInfos: StandaloneOptionsInfos
+): StandaloneOptions {
+  const optionValues: Record<string, boolean | string> = {};
+  for (const [optionName, info] of Object.entries(optionsInfos)) {
+    const parser = info.parser || optionEnabled;
+    optionValues[optionName] = parser(camelCaseToSnakeCase(optionName));
+  }
+  return (optionValues as unknown) as StandaloneOptions;
+}
+
+// This is just a cast in one place.
+function optionsToRecord(options: StandaloneOptions) {
+  return (options as unknown) as Record<string, boolean | string>;
+}
+
+const options = getOptionsInfoFromSearchParameters(optionsInfo);
```

Comment:
Looks like options are snapshotted on page load, so we should have a note that changing the options requires reload

Reply:
Doh! You're right. I kind of mindlessly added them. I'll change it to not be snapshotted.

Reply:
Ah, I see refactoring for them to be live is probably too much work. Just added the note as you suggested.